src/webview_provider/queryResultPanel.ts

```diff
@@ -679,6 +683,17 @@ export class QueryResultPanel extends AltimateWebviewProvider {
       );
       return;
     }
+    const queryHistoryCurrentSize = getStringSizeInMb(
```

Comment:
is there a need to get current size so precisely? as `getStringSizeInMb` might be time consuming function for bigger objects. i think `string.length / (1024 * 1024)` would suffice at-least for this use case.

Reply:
Initially i had something simple like that. but modified it to just make sure we have right size history and limit to 3MB. Since we are limiting to 3MB, this function should not impact performance much

Reply:
@saravmajestic Will this be a blocking call? Can you keep the value updated after query preview so that if doesn't get in the way when new queries are previewed?